src/databricks/labs/ucx/source_code/jobs.py

```diff
@@ -389,51 +391,38 @@ def _lint_job(self, job: jobs.Job) -> list[JobProblem]:
-        dependency: Dependency = WorkflowTask(self._ws, task, job)
+        start_dep: Dependency = WorkflowTask(self._ws, task, job)
         # we can load it without further preparation since the WorkflowTask is merely a wrapper
-        container = dependency.load(self._path_lookup)
+        container = start_dep.load(self._path_lookup)
         assert isinstance(container, WorkflowTaskContainer)
         session_state = CurrentSessionState(
             data_security_mode=container.data_security_mode,
             named_parameters=container.named_parameters,
             spark_conf=container.spark_conf,
             dbr_version=container.runtime_version,
         )
-        graph = DependencyGraph(dependency, None, self._resolver, self._path_lookup, session_state)
+        graph = DependencyGraph(start_dep, None, self._resolver, self._path_lookup, session_state)
         problems = container.build_dependency_graph(graph)
         if problems:
             for problem in problems:
                 source_path = self._UNKNOWN if problem.is_path_missing() else problem.source_path
                 yield LocatedAdvice(problem.as_advisory(), source_path)
             return
-        for dependency in graph.root_dependencies:
-            root = dependency.path  # since it's a root
-            yield from self._lint_one(task, dependency, graph, root, session_state, linted_paths)
+        migration_index = self._migration_index
 
-    def _lint_one(
-        self,
-        task: jobs.Task,
-        dependency: Dependency,
-        graph: DependencyGraph,
-        root_path: Path,
-        session_state: CurrentSessionState,
-        linted_paths: set[Path],
-    ) -> Iterable[LocatedAdvice]:
-        if dependency.path in linted_paths:
-            return
-        linted_paths.add(dependency.path)
-        logger.info(f'Linting {task.task_key} dependency: {dependency}')
-        if dependency.path.is_file() or is_a_notebook(dependency.path):
-            inherited_tree = graph.root.build_inherited_tree(root_path, dependency.path)
-            ctx = LinterContext(self._migration_index, session_state)
-            path_lookup = self._path_lookup.change_directory(dependency.path.parent)
-            # FileLinter will determine which file/notebook linter to use
-            linter = FileLinter(ctx, path_lookup, session_state, dependency.path, inherited_tree)
-            for advice in linter.lint():
-                yield LocatedAdvice(advice, dependency.path)
-        maybe_graph = graph.locate_dependency(dependency.path)
-        # problems have already been reported while building the graph
-        if maybe_graph.graph:
-            child_graph = maybe_graph.graph
-            for child_dependency in child_graph.local_dependencies:
-                yield from self._lint_one(task, child_dependency, child_graph, root_path, session_state, linted_paths)
+        class LintingWalker(DependencyGraphWalker[LocatedAdvice]):
```

Comment:
can we make this class top-level in this file?

Reply:
done